StatsDownload/StatsDownload.Core/Implementations/Tested/DownloadProvider.cs

```diff
@@ -27,11 +27,18 @@ public void DownloadFile(FilePayload filePayload)
         {
             loggingService.LogVerbose($"Attempting to download file: {dateTimeService.DateTimeNow()}");
 
-            using (IWebClient webClient = webClientFactory.Create())
+            IWebClient webClient = null;
+
+            try
             {
+                webClient = webClientFactory.Create();
                 SetUpFileDownload(filePayload, webClient);
                 DownloadFile(filePayload, webClient);
-                webClient.SslPolicyOverride = null;
+            }
+            finally
+            {
+                webClient?.Dispose();
+                webClientFactory.Release(webClient);
```

Comment:
Does this not need the NULL check also?

Reply:
if the create failed and did not return then this would prevent another exception being thrown
its not like this anymore, only calling release now